src/connection.ts

```diff
@@ -185,28 +182,12 @@ export class Connection extends JSForceConnection {
    * Retrieves the highest api version that is supported by the target server instance.
    */
   public async retrieveMaxApiVersion(): Promise<string> {
-    // errors go to stdout from within jsforce (maybe even further down the stack) if org's url doesn't exist.  (not a 404 from the domain but a DNS ENOTFOUND)
-    // We need to verify that the org can be connected to BEFORE asking about it's api version.  This will throw on DNS errors
-    if (!this.options.connectionOptions?.instanceUrl) {
-      throw new Error('Connection has no instanceUrl');
-    }
-    const resolver = await MyDomainResolver.create({
-      url: new URL(this.options.connectionOptions.instanceUrl),
-      timeout: Duration.minutes(0),
-    });
-    const ipAddress = await resolver.resolve();
-
-    if (ipAddress) {
-      type Versioned = { version: string };
-      const versions = (await this.request(
-        `${this.options.connectionOptions.instanceUrl}/services/data`
-      )) as Versioned[];
-      this.logger.debug(`response for org versions: ${versions}`);
-      const max = ensure(maxBy(versions, (version: Versioned) => version.version));
-      return max.version;
-    }
-
-    throw new Error();
+    await this.isResolvable();
+    type Versioned = { version: string };
+    const versions = (await this.request(`${this.instanceUrl}/services/data`)) as Versioned[];
+    this.logger.debug(`response for org versions: ${versions}`);
```

Comment:
I think this will output something like, `response for org versions: [object Object], [object Object]...`  Maybe we just want to output the max?

Reply:
now, it logs all versions as an array of string